parlai/core/torch_agent.py

```diff
@@ -1930,19 +1996,23 @@ def self_observe(self, self_message: Message) -> None:
 
         raise RuntimeError("Unexpected case in self_observe.")
 
-    def _validate_observe_invariants(self):
+    def _validate_observe_invariants(self, observation: Message = None):
         """
         Check that we properly called self_observe after the last batch_act.
         """
-        if self.__expecting_to_reply:
+        if not self.allow_multiple_observe and self.__expecting_to_reply:
```

Comment:
Look like this should take care of backward compatibility. But have tests that make sure it is (if there aren't any already).

Reply:
Sounds great I'll add unit tests once I have silence and teachers, etc. figured out